codec/amino.go

```diff
@@ -32,9 +32,17 @@ func New() *Codec {
 // RegisterEvidences registers Tendermint evidence types with the provided Amino
 // codec.
 func RegisterEvidences(cdc *Codec) {
-	tmtypes.RegisterEvidences(cdc.Amino)
+	cdc.Amino.RegisterInterface((*tmtypes.Evidence)(nil), nil)
+	cdc.Amino.RegisterConcrete(&tmtypes.DuplicateVoteEvidence{}, "tendermint/DuplicateVoteEvidence", nil)
+	cdc.Amino.RegisterConcrete(&tmtypes.ConflictingHeadersEvidence{}, "tendermint/ConflictingHeadersEvidence")
+	cdc.Amino.RegisterConcrete(&tmtypes.LunaticValidatorEvidence{}, "tendermint/LunaticValidatorEvidence")
+	cdc.Amino.RegisterConcrete(&tmtypes.PotentialAmnesiaEvidence{}, "tendermint/PotentialAmnesiaEvidence")
+	cdc.Amino.RegisterConcrete(&tmtypes.AmnesiaEvidence{}, "tendermint/AmnesiaEvidence")
```

Comment:
@alexanderbez @zmanian this is what I was thinking. The sdk call `node.Block` and then encodes it. This mimics what was registered prior

Reply:
TBH I really don't follow the issue here. I don't understand why we need to encode or register anything?

Reply:
the way i see it is this: https://github.com/cosmos/cosmos-sdk/blob/e6a56226a88ec4e8761b1388a3a666af627d9e2e/client/rpc/block.go#L65-L65 you receive the block and need to json encode it. To json encode you need to register the evidence interface.

Reply:
Why is the block amino encoded?

Reply:
![](https://media2.giphy.com/media/ogb8RQdu8zQyc/giphy.gif)

Reply:
not sure if the wrapper of the tendermint RPC has been worked on for grpc migration yet, so it still has the legacy fromat

Reply:
will merge this as is but we can continue the discussion here and I can amend the base pr, if it needs to be